yarn-project/simulator/src/avm/opcodes/memory.ts

```diff
@@ -152,9 +153,14 @@ export class Mov extends Instruction {
   }
 
   async execute(context: AvmContext): Promise<void> {
-    const a = context.machineState.memory.get(this.srcOffset);
+    const [srcOffset, dstOffset] = Addressing.fromWire(this.indirect).resolve(
+      [this.srcOffset, this.dstOffset],
+      context.machineState.memory,
+    );
 
-    context.machineState.memory.set(this.dstOffset, a);
+    const a = context.machineState.memory.get(srcOffset);
+
+    context.machineState.memory.set(dstOffset, a);
```

Comment:
It would be lovely if we could somehow make this automatic for all offset arguments so they don't need to be explicitly resolved like this for every instruction with offsets :thinking: But maybe it's just not feasible?